mocking/file-logger/go/logger.go

```
@@ -0,0 +1,93 @@
+package main
+
+import (
+	"time"
+)
+
+const (
+	DefaultWeekendFileName = "weekend.txt"
+)
+
+type FileLogger struct {
+	storage FileStorage
+	clock   clock
+}
+
+func (FileLogger) formatWeekdayFile(currentTime *time.Time) string {
+	return "log" + currentTime.Format("20060102") + ".txt"
+}
+
+func (FileLogger) formatPreviousWeekendFile(modTime *time.Time) string {
+	return "weekend-" + modTime.Format("20060102") + ".txt"
+}
+
+func (FileLogger) wasFileModifiedThisWeekend(currentTime, modTime *time.Time) bool {
+	var currentWeekBeginning time.Time
+
+	if currentTime.Weekday() == time.Saturday {
+		year, month, day := currentTime.Date()
+		currentWeekBeginning = time.Date(year, month, day, 0, 0, 0, 0, time.UTC)
+	} else {
+		year, month, day := currentTime.Date()
+		currentWeekBeginning = time.Date(year, month, day-1, 0, 0, 0, 0, time.UTC)
+	}
+
+	// check if file was created during this weekend
+	if modTime.After(currentWeekBeginning) {
+		return true
+	}
+
+	return false
+}
+
+func (l *FileLogger) getFileName() string {
+	currentTime := l.clock.Now()
+
+	if !IsWeekend(currentTime) {
+		return l.formatWeekdayFile(&currentTime)
+	}
+
+	if !l.storage.FileExists(DefaultWeekendFileName) {
+		return DefaultWeekendFileName
+	}
+
+	modTime, err := l.storage.FileModificationTime(DefaultWeekendFileName)
+
+	if err != nil {
+		panic("Couldn't get the modification time from " + DefaultWeekendFileName)
```

unless in special cases, I'd just return the error up, wrapped if needed. panics are to be avoided in Go